main.cxx

```diff
@@ -0,0 +1,71 @@
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <string_view>
+#include <vector>
+
+template<class Iterator>
+std::vector<int> scoresBetween(const Iterator &, const Iterator &, const Iterator &, const Iterator &);
+std::string lcs(const std::string_view &, const std::string_view &);
+
+int main() {
+	int T;
+	std::ios::sync_with_stdio(false);
+
+	std::cin >> T;
+	while (T--) {
+		std::string X, Y;
+		std::cin >> X >> Y;
+
+		std::cout << lcs(X, Y) << std::endl;
+	}
+
+	return 0;
+}
+
+std::string lcs(const std::string_view &x, const std::string_view &y) {
+	if (x.length() == 0 || y.length() == 0) return "";
+	if (x.length() == 1 && y.find_first_of(x) != std::string_view::npos) return { x[0] };
+	if (y.length() == 1 && x.find_first_of(y) != std::string_view::npos) return { y[0] };
+
+	auto mid = x.length() / 2;
+	auto result = scoresBetween(
+			x.cbegin(), x.cbegin() + mid,
+			y.cbegin(), y.cend()
+			);
+	auto resultReverse = scoresBetween(
+			x.crbegin(), x.crbegin() + (x.length() - mid),
+			y.crbegin(), y.crend()
+			);
```

Comment:
I prefer `topScores` and `bottomScores` rather than `result` and `resultReverse`.